activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb

```diff
@@ -117,6 +123,25 @@ def visit_DropCheckConstraint(name)
           "DROP CONSTRAINT #{quote_column_name(name)}"
         end
 
+        def visit_ExclusionConstraintDefinition(o)
```

Comment:
Since this syntax is postgres specific, I think it should live here: https://github.com/rails/rails/blob/main/activerecord/lib/active_record/connection_adapters/postgresql/schema_creation.rb

Reply:
@ghiculescu I did my best to mimic the patterns already established for check constraints, which have their SQL defined in the abstract class rather than the PG-specific one. I'm happy to move this, but that would result in some inconsistency, which I'm not sure is worth it. I'd feel more comfortable hearing from someone on the Core Team about where this should go before moving it away from the similar check constraint code!